lib/galaxy/jobs/__init__.py

```diff
@@ -1181,7 +1181,7 @@ def galaxy_url(self):
         return self.get_destination_configuration("galaxy_infrastructure_url")
 
     def get_job(self) -> model.Job:
-        return self.sa_session.get(Job, self.job_id)
+        return self.sa_session.get(Job, self.job_id)  # type:ignore[return-value]
```

Comment:
This is a lot of type ignores in this PR. This one feels of an easier variety....

Can we have a helper utility that does that and does a typing.cast to prevent the typing ignore here.

```
from typing import cast, TypeVar

T = TypeVar('T')

def load_from_session(session, clazz: T,  id: int) -> T:
    val = session.get(clazz, id)
    return cast(clazz, val)
```

I don't think we want a typing ignore every place we are doing a session.get.

It looks like a lot of these are scalars() which is untyped... maybe we could do a typed version of scalars that does the cast also? I am not sure if that makes sense but I would love to centralize the ignores if possible. Random application code shouldn't have to have these ignores in there I think.

It looks like a lot of these ignores are ``type:ignore[return-value]`` also - I would prefer setting a variable of the right type and ignoring the problem there and returning "the right thing" without a type ignore. Maybe that is silly - but to me that would feel like the hacks are more enclosed in their blackboxes.

Reply:
[This edit was part of the SA2.0 PR (#17180)] 

If I'm not mistaken, this type-ignore is there to ignore the None value that the `session.get` may return (by definition of `session.get`). The non-None value should be typed correctly (`model.Job`) - and that works as expected in most other places, I think. However, in this particular case we *know* it's never returning None, because the method is defined on the `JobWrapper` class which always has a `job_id` (it takes a Job arg in the constructor and assigns its `id` attr to `self.job_id`; and that Job should always belong to a session, I assume). So, I think, the type-ignore here is a legitimate way to help mypy make the right call based on the application logic that it cannot interpret on its own.

The other type-ignores (for return-value) vary. In some cases they point to a legitimate problem: a value can be None, which we don't always check before calling methods on that value (I don't remember which ones are like that). My intention was to (1) type-ignore those cases for now since those typing errors were not directly caused by a change introduced in the PR, but by the PR adding typing to the model (required by SA2.0), and that typing causing ripple effects throughout the code base: i.e., mypy type-checks a lot more now.  And then, after SA2.0 is part of dev, (2) address each newly added type-ignore - in which case I would either refactor to fix the problem (if one exists), and/or use your suggested approach for the case when `session.get` returns an unexpected type. Would that be OK?

Reply:
The following should solve the problem with `None`:

```suggestion
        job_id = self.sa_session.get(Job, self.job_id)
        assert job_id
        return job_id
```

Reply:
@nsoranzo thanks, I'll apply this to the SA20 PR too.